packages/core/src/modules/proofs/protocol/v2/V2ProofService.ts

```diff
@@ -0,0 +1,831 @@
+      associatedRecordId: proofRecord.id,
+      messageClass: V2ProposalPresentationMessage,
+    })
+
+    if (!proposal) {
+      return false
+    }
+
+    const request = await this.didCommMessageRepository.findAgentMessage({
+      associatedRecordId: proofRecord.id,
+      messageClass: V2RequestPresentationMessage,
+    })
+
+    if (!request) {
+      throw new AriesFrameworkError(`Expected to find a request message for ProofRecord with id ${proofRecord.id}`)
+    }
+
+    const proposalAttachments = proposal.getAttachmentFormats()
+    const requestAttachments = request.getAttachmentFormats()
+
+    const equalityResults = []
+    for (const attachmentFormat of proposalAttachments) {
+      const service = this.getFormatServiceForFormat(attachmentFormat.format)
+      equalityResults.push(service?.proposalAndRequestAreEqual(proposalAttachments, requestAttachments))
+    }
+
+    return equalityResults.every((x) => x === true)
+  }
+
+  public async shouldAutoRespondToPresentation(proofRecord: ProofRecord): Promise<boolean> {
+    const request = await this.didCommMessageRepository.getAgentMessage({
+      associatedRecordId: proofRecord.id,
+      messageClass: V2RequestPresentationMessage,
+    })
+
+    return request.willConfirm
+  }
+
+  public async createProofRequestFromProposal(options: {
+    formats: {
+      indy?: {
+        proofRecord: ProofRecord
+      }
+      jsonLd?: never
+    }
+    config?: { indy?: { name: string; version: string; nonce?: string }; jsonLd?: never }
+  }): Promise<{
+    indy?: ProofRequest
+    jsonLd?: never
+  }> {
+    let result = {}
+
+    for (const [format, value] of Object.entries(options.formats)) {
+      const service = this.formatServiceMap[format]
+
+      const indyFormat = options.formats.indy
+      const indyConfig = options.config?.indy
+
+      if (!indyFormat) {
+        throw new AriesFrameworkError('Indy format must be provided')
+      }
+
+      if (!indyConfig) {
+        throw new AriesFrameworkError('Indy config must be provided')
+      }
+
+      const proofRecordId = indyFormat.proofRecord.id
+      const proposalMessage = await this.didCommMessageRepository.findAgentMessage({
+        associatedRecordId: proofRecordId,
+        messageClass: V2ProposalPresentationMessage,
+      })
+
+      if (!proposalMessage) {
+        throw new AriesFrameworkError(`Proof record with id ${proofRecordId} is missing required presentation proposal`)
+      }
+
+      result = {
+        ...result,
+        ...(await service.createProofRequestFromProposal({
+          formats: {
+            indy: {
+              presentationProposal: proposalMessage?.proposalsAttach[0],
+            },
+          },
+          config: { indy: indyConfig, jsonLd: undefined },
+        })),
+      }
+    }
+    return result
+  }
+
+  public async findRequestMessage(options: { proofRecord: ProofRecord }): Promise<AgentMessage | null> {
```

Comment:
This requires all of the required attributes of `ProofRecord` even though only the `id` attribute is used. This means someone needs to know the state, threadId and protocolVersion to use this method where they should only need to know the id

Reply:
best work around for this is to query the proofRecord right before using this method which seems fine